applications/FluidDynamicsApplication/custom_strategies/strategies/fs_strategy.h

```diff
@@ -733,32 +633,22 @@ class FSStrategy : public SolvingStrategy<TSparseSpace,TDenseSpace,TLinearSolver
     virtual void ComputeSplitOssProjections(ModelPart& rModelPart)
     {
         array_1d<double,3> Out = ZeroVector(3);
-
-#pragma omp parallel
-        {
-            ModelPart::NodeIterator NodesBegin;
-            ModelPart::NodeIterator NodesEnd;
-            OpenMPUtils::PartitionedIterators(rModelPart.Nodes(),NodesBegin,NodesEnd);
-
-            for ( ModelPart::NodeIterator itNode = NodesBegin; itNode != NodesEnd; ++itNode )
-            {
-                itNode->FastGetSolutionStepValue(CONV_PROJ) = ZeroVector(3);
-                itNode->FastGetSolutionStepValue(PRESS_PROJ) = ZeroVector(3);
-                itNode->FastGetSolutionStepValue(DIVPROJ) = 0.0;
-                itNode->FastGetSolutionStepValue(NODAL_AREA) = 0.0;
-            }
+        const int n_nodes = rModelPart.NumberOfNodes();
+        const int n_elems = rModelPart.NumberOfElements();
+
+#pragma omp parallel for
+        for (int i_node = 0; i_node < n_nodes; ++i_node) {
+            auto it_node = rModelPart.NodesBegin() + i_node;
+            it_node->FastGetSolutionStepValue(CONV_PROJ) = CONV_PROJ.Zero();
+            it_node->FastGetSolutionStepValue(PRESS_PROJ) = PRESS_PROJ.Zero();
+            it_node->FastGetSolutionStepValue(DIVPROJ) = 0.0;
+            it_node->FastGetSolutionStepValue(NODAL_AREA) = 0.0;
```

Comment:
probably not for this PR, but do those need to be historical?

Reply:
I don't think so. In any case, this is a delicate change that would require a PR on its own.

Reply:
> this is a delicate change that would require a PR on its own.

absolutely!

Reply:
There are also non historical vars (of the same names) used in the strategy :-)